pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -176,6 +176,9 @@ func ApplyTaskResults(targets PipelineRunState, resolvedResultRefs ResolvedResul
 			pipelineTask.Params = replaceParamValues(pipelineTask.Params, stringReplacements, arrayReplacements, objectReplacements)
 			if pipelineTask.IsMatrixed() {
 				pipelineTask.Matrix.Params = replaceParamValues(pipelineTask.Matrix.Params, stringReplacements, nil, nil)
+				for i := range pipelineTask.Matrix.Include {
+					pipelineTask.Matrix.Include[i].Params = replaceParamValues(pipelineTask.Matrix.Include[i].Params, stringReplacements, nil, nil)
```

Comment:
curious why array replacements and object replacements aren't supported here? it may be worth adding a comment

Reply:
we do not yet (but want to) support array results in matrix.params - @abayer was implementing this in https://github.com/tektoncd/pipeline/pull/6056

object results don't make sense in the matrix design - see relevant discussion in https://github.com/tektoncd/pipeline/pull/6056#discussion_r1088368159


for matrix.include.params, I'd expect string results only because it takes parameters of type string only - https://github.com/tektoncd/community/blob/main/teps/0118-matrix-with-explicit-combinations-of-parameters.md#api-change

Reply:
Added a comment